tools/dataset/preprocess_mpi_inf_3dhp.py

```diff
@@ -0,0 +1,362 @@
+                |-- train
+                    |-- S1
+                        |-- Seq1
+                        |-- Seq2
+                    |-- S2
+                    |-- ...
+                |-- test
+                    |-- TS1
+                    |-- TS2
+                    |-- ...
+        out_dir: Directory to save annotation file.
+    """
+    _imgnames = []
+    _centers = []
+    _scales = []
+    _joints_2d = []
+    _joints_3d = []
+    cameras = {}
+
+    img_dir = join(out_dir, 'images')
+    os.makedirs(img_dir, exist_ok=True)
+    annot_dir = join(out_dir, 'annotations')
+    os.makedirs(annot_dir, exist_ok=True)
+
+    for subj in train_subjects:
+        for seq in train_seqs:
+            seq_path = join(data_root, 'train', f'S{subj}', f'Seq{seq}')
+            num_frames = train_frame_nums[(subj, seq)]
+
+            # load camera parametres
+            camera_file = join(seq_path, 'camera.calibration')
+            with open(camera_file, 'r') as fin:
+                lines = fin.readlines()
+                for cam in train_cams:
+                    K = [float(s) for s in lines[cam * 7 + 5][11:-2].split()]
+                    f = np.array([[K[0]], [K[5]]])
+                    c = np.array([[K[2]], [K[6]]])
+                    RT = np.array(
+                        [float(s) for s in lines[cam * 7 + 6][11:-2].split()])
+                    RT = np.reshape(RT, (4, 4))
+                    R = RT[:3, :3]
+                    # convert unit from millimeter to meter
+                    T = RT[:3, 3:] * 0.001
+                    size = [
+                        float(s) for s in lines[cam * 7 + 3][14:-2].split()
+                    ]
+                    w, h = size
+                    cam_param = dict(
+                        R=R, T=T, c=c, f=f, w=w, h=h, name=f'train_cam_{cam}')
+                    cameras[f'S{subj}_Seq{seq}_Cam{cam}'] = cam_param
+
+            # load annotations
+            annot_file = os.path.join(seq_path, 'annot.mat')
+            annot2 = loadmat(annot_file)['annot2']
+            annot3 = loadmat(annot_file)['annot3']
+            for cam in train_cams:
+                # load 2D and 3D annotations
+                joints_2d = np.reshape(annot2[cam][0][:num_frames],
+                                       (num_frames, 28, 2))[:, joints_17]
+                joints_3d = np.reshape(annot3[cam][0][:num_frames],
+                                       (num_frames, 28, 3))[:, joints_17]
+                joints_3d = joints_3d * 0.001
+                centers, scales, joints_2d, joints_3d = get_annotations(
+                    joints_2d, joints_3d)
+                _centers.append(centers)
+                _scales.append(scales)
+                _joints_2d.append(joints_2d)
+                _joints_3d.append(joints_3d)
+
+                # extract frames from video
+                video = join(seq_path, 'imageSequence', f'video_{cam}.avi')
+                cap = cv2.VideoCapture(video)
```

Comment:
Could you please try mmcv video IO interface [VideoReader](https://github.com/open-mmlab/mmcv/blob/master/docs/video.md) and mmcv.track_iter_progress (to replace tqdm) here?